### ISSUE TITLE
[BUG]: messageFileEmpty not showing

### ISSUE THREAD

**Issue by assadnazar** [phalcon/cphalcon#14928]
File Validation Empty custom message is not being shown.
```php
$facilityimage = new File('facilityimage');

$facilityimage->addValidators([
    //new FileValidator()
    new FileValidator([
        'maxSize' => '0.5M',
        'messageSize' => 'Your image is too big. Max file size: 500KB',
        'allowedTypes' => array('image/jpeg', 'image/png'),
        'messageType' => 'Your image must be a JPEG or PNG file',
        'allowEmpty' => false,
        'messageFileEmpty' => 'No image uploaded.'
    ])
]);

$this->add($facilityimage);
```

Only default message is being shown."Field facilityimage must not be empty"

**Comment by Izopi4a** [comment]
Hi, 

I think you are not passing the correct params to the validate method.

working example:

`form`

```
use Phalcon\Forms\Element\Text;
use Phalcon\Forms\Element\File;
use Phalcon\Forms\Form;
use Phalcon\Validation\Validator\PresenceOf;
use Phalcon\Validation\Validator\File as FileValidator;


class FormOne extends Form
{
    /**
     * @param null $entity
     * @param array $options
     */
    public function initialize($entity = null, array $options = [])
    {

        $name = new Text('name', [
            'placeholder' => 'Name',
            'class'=> 'form-control'
        ]);

        $name->addValidators([
            new PresenceOf([
                'message' => 'The name is required',
            ]),
        ]);

        $this->add($name);

        $facilityimage = new File('image1');


        $facilityimage->addValidators([
            //new FileValidator()
            new FileValidator([
                'maxSize' => '0.5M',
                'messageSize' => 'Your image is too big. Max file size: 500KB',
                'allowedTypes' => array('image/jpeg', 'image/png'),
                'messageType' => 'Your image must be a JPEG or PNG file',
                'allowEmpty' => false,
                'messageFileEmpty' => 'No image uploaded.'
            ])
        ]);

        $this->add($facilityimage);

    }
}
```

`controller`

```
public function test14928Action() {

        $this->view->setTemplateBefore('public');

        $form = new FormOne();

        $this->view->setVar("form", $form);

        if ($this->request->isPost()) {

            echo "<pre>";
            var_dump($form->isValid(array_merge($_POST, $_FILES))); // <----
            echo "</pre>";
        }
    }
```

`view`

```
{% set form = this.view.getVar("form") %}


<form id="form" action="/index/test14928" method="post" enctype="multipart/form-data">
    <div class="form-group">
        {{ form.render("name") }}
    </div>
    <div class="form-group">
        {{ form.render("image1") }}
    </div>
    <button type="submit" class="btn btn-primary">Submit</button>
</form>
```

**Comment by StudioMaX** [comment]
In version 3.4.x `Phalcon\Validation\Validator\File` was the standard validator, which also has the `isAllowEmpty` method. If this method was present, it was called during `Validation::preChecking` before executing the file validator itself. 
https://github.com/phalcon/cphalcon/blob/v4.1.2/phalcon/Validation.zep#L512-L517

In 4.x `Phalcon\Validation\Validator\File` was changed to extend `AbstractValidatorComposite`. Both of them doesn't contain `isAllowEmpty`, but is still exists in `Phalcon\Validation\Validator\File\AbstractFile`. But `Validation::preChecking` doesn't see this method, since it know nothing about `AbstractFile`. `AbstractFile` is needed only for submodules (`File\Min`, `File\Max`, etc.), but not for `Phalcon\Validation\Validator\File`. So `isAllowEmpty` isn't executed at all.

**Comment by BeMySlaveDarlin** [comment]
@StudioMaX, your assumption wrong in case of `isAllowEmpty` isn't executed.

It is executed, but the message isn't reassigend due to that changes you mentioned.
Due to `Phalcon\Validation\Validator\File` is composite, it builds each of submodules during initialization, but legacy messages isn't passed by, so it throws default message.

Did a fix in #15769

**Comment by niden** [comment]
Resolved in https://github.com/phalcon/cphalcon/pull/15769
